src/pages/api/auth/[...nextauth].ts

```diff
@@ -122,13 +131,19 @@ export default NextAuth({
 		 *
 		 * ref: https://next-auth.js.org/configuration/callbacks#session-callback
 		 */
-		async session({ session, token }) {
+		async session({
+			session,
+			token,
+		}: {
+			token: JWT
+			session: NextAuthSession
+		}): Promise<SessionData> {
 			return {
 				...session,
-				accessToken: token.access_token,
-				id: token.sub,
-				user: { ...session.user, nickname: token.nickname },
-				error: token.error,
+				accessToken: token.access_token as TokenSet['access_token'],
```

Comment:
Again I had a hard time typing this. I can't set the token to `JWT` on the function itself because next auth requires it to be a JWT, so I have to cast as the `TokenSet`. Maybe I can just add a @TSFixMe to address later.